### PR TITLE
Bare Metal provisioning and setup

### PR DESCRIPTION
**Pull request by johnstudarus**
Support for c2.medium provisioning via Terraform including setting up disk RAID and network MTU for OpenStack.

Disks are mounted at /openstack.
MTU set for 8990 on non-Internet interfaces.

A basic openstack_user_config.yml is generated via Terraform from the provisioning information (IP addresses and hostnames).

### CODE REVIEW

File: terraform/vars.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+variable "control_count" {
+  description = "Number of OpenStack control nodes to deploy"
+  default = "1"
+}
+
+variable "operating_system" {
+  description = "Operating System to install across nodes"
+  default = "ubuntu_16_04"
```
Comment (pabelanger) [inline terraform/vars.tf]: Can we change this to 18.04, since that is the latest LTS.

File: terraform/id_rsa.pub [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC6d3lJdnWzH8l4Pf55TeTfAy/3GUaI0BYdwZljpNaX/L5hVvOWVPaUIBuxuTylf5MBxAuotv8ZxGFkUaWTXJaLzRUZ8l5d8jztdft58v4rHcTthdTafKC50JPw+WBhHgWLRxddvcxW/ou6R5X9VLKZ/wCwTCwQgZEecf5s3xXQjvFN2urI8nrauVMLypDfl0hwaPylf0FBfTApdscAFQY+RAvPU1Sb1Mp2UVzB3+bblGHNa5IE/OJDujnisEDAM1hbkyyRWQWzh7yXqML9G5LNF0E5KYAioAgUjmPKnik4B/aXQhDEip0mVpuhM8iPvOVGW+z65Q89hq1sIbLJnVRd studarus@shell
```
Comment (pabelanger) [inline terraform/id_rsa.pub:1]: What user does this get installed into on the remote server? I know OSA has some logic in to manage keys also, maybe we should just do that by default.

**Review by pabelanger — COMMENTED** [review]
Not really sure how most of this works, but left a comment.

Also, can we rebase the 33 commits into a single against master? This will help to limit the number of new commits we'll add to the repo.

File: terraform/id_rsa.pub (hunk 2 above)
Comment (johnstudarus) [reply terraform/id_rsa.pub:1]: That sets up the keys for root. We can set up an alternate user if desired.

**Comment by johnstudarus** [comment]
Restarting with a new pull request with just a single commit...

**Pull request closed without merging**